lib/compile.js

```diff
@@ -105,6 +106,23 @@ function compileLoad(loader, load, compileOpts, cache) {
     if (format == 'cjs')
       mappedLoad.source = mappedLoad.source.replace(hashBangRegEx, '');
     return Promise.resolve()
+    .then(function () {
+      var sourceMap = sourceMappingURL.getFrom(mappedLoad.source);
```

Comment:
I think skipping on `metadata.sourceMap` already set makes the most sense. Plugins should handle extracting sourceMappingURL comments themselves if they want to support transitivity, as we can't just "add it on" here for a plugin that has already transpiled.